src/cli.rs

```diff
@@ -164,6 +164,10 @@ pub struct Ic {
     #[clap(env, long, default_value = "5")]
     pub ic_request_retries: usize,
 
+    /// Maximum number of requests that can be executed in parallel.
+    #[clap(env, long, default_value = "2000")]
```

Comment:
I think the option is not needed. Just hardcode it to 20k so that we do not hit it

Reply:
done